tests/test/util/test_queue.cpp

```diff
@@ -147,4 +163,15 @@ TEST_CASE("Test queue timeout must be positive", "[util]")
     q.enqueue(10);
     REQUIRE_THROWS(q.dequeue(timeoutValueMs));
 }
+
+TEST_CASE("Test fixed capacity queue blocks if queue is full", "[util]")
+{
+    FixedCapIntQueue q(2);
+
+    q.enqueue(1);
+    q.enqueue(2);
+
+    // Enqueue with a short timeout so the operation fails quickly
+    REQUIRE_THROWS(q.enqueue(100));
```

Comment:
I'm not sure this test is really doing what it claims to do, it only shows that an unspecified error is thrown if we enqueue above the capacity. While this is a valid test, it could be improved by checking that the exception thrown is actually a timeout exception.

More importantly, we still need a test that checks that the queue blocks and eventually does return if it's full and a dequeue happens sufficiently quickly. Without this, there's some important locking and condition variable logic that's not checked (and we know from experience that this is where concurrency bugs hide).